src/HttpClient/HttpPluginClientBuilder.php

```diff
@@ -12,26 +12,26 @@
 use Http\Client\Common\HttpMethodsClient;
 use Http\Client\Common\Plugin;
 use Http\Client\Common\PluginClient;
-use Http\Discovery\HttpClientDiscovery;
-use Http\Discovery\MessageFactoryDiscovery;
-use Http\Message\RequestFactory;
+use Http\Discovery\Psr17FactoryDiscovery;
+use Http\Discovery\Psr18ClientDiscovery;
 use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
 
 class HttpPluginClientBuilder
 {
     /** @var ClientInterface */
     private $httpClient;
     /** @var HttpMethodsClient|null */
     private $pluginClient;
-    /** @var RequestFactory */
+    /** @var RequestFactoryInterface */
     private $requestFactory;
     /** @var Plugin[] */
     private $plugins = [];
 
-    public function __construct(ClientInterface $httpClient = null, RequestFactory $requestFactory = null)
+    public function __construct(ClientInterface $httpClient = null, RequestFactoryInterface $requestFactory = null)
```

Comment:
This is a braking change. Any way we can introduce this in a none breaking way e.g. have the method accept `RequestFactory|RequestFactoryInterface`? Otherwise, we'd have to release this as a new major version, where we should probably remove several of the deprecated methods too but would prefer adding this none breaking first with maybe a deprecation message.